serializers/src/main/java/io/pravega/schemaregistry/schemas/ProtobufSchema.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.schemas;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.DescriptorProtos;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.Message;
+import com.google.protobuf.Parser;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.contract.data.SerializationFormat;
+import lombok.Data;
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Container class for protobuf schema.
+ * Protobuf schemas are represented using {@link com.google.protobuf.DescriptorProtos.FileDescriptorSet}. 
+ *
+ * @param <T> Type of element. 
+ */
+@Data
+public class ProtobufSchema<T extends Message> implements SchemaContainer<T> {
+    @Getter
+    private final Parser<T> parser;
+    @Getter
+    private final DescriptorProtos.FileDescriptorSet descriptorProto;
+
+    private final SchemaInfo schemaInfo;
+
+    private ProtobufSchema(String name, Parser<T> parser, DescriptorProtos.FileDescriptorSet fileDescriptorSet) {
+        this.parser = parser;
+        this.descriptorProto = fileDescriptorSet;
+        this.schemaInfo = new SchemaInfo(name, SerializationFormat.Protobuf, getSchemaBytes(), ImmutableMap.of());
+    }
+
+    @SneakyThrows
+    private ProtobufSchema(SchemaInfo schemaInfo) {
+        DescriptorProtos.FileDescriptorSet fileDescriptorSet = DescriptorProtos.FileDescriptorSet.parseFrom(schemaInfo.getSchemaData());
+
+        this.parser = null;
+        this.descriptorProto = fileDescriptorSet;
+        this.schemaInfo = schemaInfo;
+    }
+
+    private ByteBuffer getSchemaBytes() {
+        return ByteBuffer.wrap(descriptorProto.toByteArray());
+    }
+
+    @Override
+    public SchemaInfo getSchemaInfo() {
+        return schemaInfo;
+    }
+
+    /**
+     * Method to generate protobuf schema from the supplied protobuf generated class and {@link DescriptorProtos.FileDescriptorSet}.
+     *
+     * @param tClass Class for code generated protobuf message.  
+     * @param fileDescriptorSet file descriptor set representing a protobuf schema. 
+     * @param <T> Type of protobuf message
+     * @return {@link ProtobufSchema} with generic type T that captures protobuf schema and parser. 
+     */
+    @SneakyThrows
```

Comment:
This is precisely the case. We have bounded the values for T by `GeneratedMessageV3`. Which means it will have the method `getDefaultInstance`.

So the exception from this line can actually never be thrown:
```        T defaultInstance = (T) tClass.getMethod("getDefaultInstance").invoke(null);
```

Reply:
I am removing it from other places wherever possible. 
So two places where i need to retain it -- AbstractPravegaSerializer and AbstractPravegaDeserializer. 
These two classes implement Serializer interface from pravega. 

Its signature doesnt declare throwing IOException so we cannot throw it from there.